app/react/Layout/DocumentsList.js

```diff
@@ -137,40 +149,29 @@ class DocumentsList extends Component {
           </div>
           {(() => {
             if (view !== 'graph') {
+              const { CollectionViewer } = this.props;
               return (
-                <RowList zoomLevel={rowListZoomLevel}>
-                  {documents.get('rows').map((doc, index) => (
-                    <Doc
-                      doc={doc}
-                      storeKey={this.props.storeKey}
-                      key={index}
-                      onClick={this.clickOnDocument}
-                      onSnippetClick={this.props.onSnippetClick}
-                      deleteConnection={this.props.deleteConnection}
-                      searchParams={this.props.search}
-                    />
-                  ))}
-                </RowList>
+                <CollectionViewer
+                  {...{
+                    rowListZoomLevel,
+                    storeKey: this.props.storeKey,
+                    clickOnDocument: this.clickOnDocument,
+                    onSnippetClick: this.props.onSnippetClick,
+                    deleteConnection: this.props.deleteConnection,
+                    onEndScroll: this.onEndScroll,
```

Comment:
As described above, consider if we really need to pass this as a full onEndScroll or if we want to just pass the loadMore function and reimplement this on the actual place where the infinite scroll is happening.

Reply:
Already answered on another comment